liblangutil/SourceReferenceFormatter.cpp

```diff
@@ -55,6 +55,28 @@ std::string SourceReferenceFormatter::formatErrorInformation(Error const& _error
 	);
 }
 
+char const* SourceReferenceFormatter::errorTextColor(Error::Severity _severity)
+{
+	switch (_severity)
+	{
+	case Error::Severity::Error: return RED;
+	case Error::Severity::Warning: return YELLOW;
+	case Error::Severity::Info: return WHITE;
+	}
+	util::unreachable();
+}
+
+char const* SourceReferenceFormatter::errorHighlightColor(Error::Severity _severity)
+{
+	switch (_severity)
+	{
+	case Error::Severity::Error: return RED_BACKGROUND;
+	case Error::Severity::Warning: return ORANGE_BACKGROUND_256;
+	case Error::Severity::Info: return GRAY_BACKGROUND;
```

Comment:
LGTM. Just one question, wouldn't this allows to display the text in white over gray background?

Reply:
Wait a second... we do have colored backgrounds for these :-D? That's weird - I mean, that's generally horrible, isn't it? But also I've never seen that in any output myself, even though the escape actually works in my terminal...

Reply:
> LGTM. Just one question, wouldn't this allows to display the text in white over gray background?

White text actually looks fine on this background. That's why I chose this color.

> Wait a second... we do have colored backgrounds for these :-D? That's weird - I mean, that's generally horrible, isn't it? But also I've never seen that in any output myself, even though the escape actually works in my terminal...

We use it in tests for highlighting parts of the source matching a particular error/warning/info. I actually find it pretty convenient - having to decode character offsets from source locations in your head sounds more horrible :) For example you can see it in `DebugWarner` output in new analysis when a test does not pass. It's not used for compilation errors by the compiler.

Since it's not used outside of tests, I could put it somewhere in test utils, but putting it together with foreground colors seemed like a better choice to me. If I ever needed it and were ever looking for it, I'd probably look here (or in `AnsiColorized.h`, but it did not seem right to make that file depend on the `Error` definition).

Reply:
Right. Just mentioned because I remembered this issue: https://github.com/ethereum/solidity/issues/14378. So I guess this PR would also fix that, no?

Reply:
I actually missed the "in isoltest" part of this PR and thought it was about regular compiler output, that's why I was confused :-).
So just to confirm, this is *not* meant to change colored solc output, but *just* to affect isoltest? (I could probably also read the code for an answer :-))

Reply:
It uses `errorColored` which in turn is used by `printExceptionInformation` and by the `CommandLineInterface` to print compiler errors in the CLI. This was why I initially thought it could be solving the issue that I mentioned above (i.e. not only affecting isoltests), but it seems that the code doesn't really change anything in the `errorColored` logic.

Reply:
> So just to confirm, this is not meant to change colored solc output, but just to affect isoltest? (I could probably also read the code for an answer :-))

Exactly. There are no behavior changes outside of tests. The non-test code is only refactored.

> So I guess this PR would also fix that, right?

No, at least not in a significant way. The issue is not limited to test output.

BTW, not sure what the "fix" would be there. I'd assume that the terminal that changes colors of various things would do that by remapping the meaning of these colors to something else. Or are there some "themable"/relative color codes that we should be using to play nice with such terminals?

Reply:
> BTW, not sure what the "fix" would be there. I'd assume that the terminal that changes colors of various things would do that by remapping the meaning of these colors to something else. Or are there some "themable"/relative color codes that we should be using to play nice with such terminals?

I'm not sure either. I know that on GNU/Linux you could use [tput](https://man7.org/linux/man-pages/man1/tput.1.html) to query for terminal-dependent information, but I don't know if a solution based on that, or on ncurses, would be reliable and cross-platform. On GNU/Linux you can do `tput colors` to get the maximum number of colors supported by the terminal according with the terminfo database, but it may not really reflect the current terminal capabilities.